src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -77,21 +83,38 @@ public function getProperties($class, array $context = [])
             return;
         }
 
-        $reflectionProperties = $reflectionClass->getProperties();
+        $propertyFlags = 0;
+        $methodFlags = 0;
+
+        if ($this->accessFlags & self::ALLOW_PUBLIC) {
+            $propertyFlags = $propertyFlags | \ReflectionProperty::IS_PUBLIC;
+            $methodFlags = $methodFlags | \ReflectionMethod::IS_PUBLIC;
+        }
+
+        if ($this->accessFlags & self::ALLOW_PRIVATE) {
+            $propertyFlags = $propertyFlags | \ReflectionProperty::IS_PRIVATE;
+            $methodFlags = $methodFlags | \ReflectionMethod::IS_PRIVATE;
+        }
+
+        if ($this->accessFlags & self::ALLOW_PROTECTED) {
+            $propertyFlags = $propertyFlags | \ReflectionProperty::IS_PROTECTED;
+            $methodFlags = $methodFlags | \ReflectionMethod::IS_PROTECTED;
+        }
+
+        $reflectionProperties = $reflectionClass->getProperties($propertyFlags);
+        $reflectionPropertiesNotFiltered = $reflectionClass->getProperties();
```

Comment:
Does it make a difference if we not call `getProperties()` twice but filter inside the loop below based on the configured flags?

Reply:
Yes, as there are used to find the correct property name given a method name. So if you have a public method name in plural and a private property name in singular, passing the filtered property would give the wrong property name (as it will not be there and cannot be linked).

Reply:
It would certainly be faster (by a light margin however), I find this better to read, but this is mainly subjective can make the change if you prefer.

Reply:
Just change to use `$reflectionProperty->getModifiers() & $propertyFlags` and works nicely, thanks for the suggestion